### DISCUSSION TITLE
bug on physical device when calling verifyPhoneNumber

### DISCUSSION THREAD

**Discussion by timurridjanovic** [invertase/react-native-firebase#5833] [Q&A]
I'm having this issue when trying to verify a phone number, but I only get the bug on a physical device. Everything works fine on a simulator.

This is the code:
```
    firebase.auth().verifyPhoneNumber(phoneNumber)
      .on('state_changed', phoneAuthSnapshot => {
        switch (phoneAuthSnapshot.state) {
          case firebase.auth.PhoneAuthState.CODE_SENT:
            resolve(phoneAuthSnapshot)
            break
          case firebase.auth.PhoneAuthState.ERROR:
            crashlytics().recordError(new Error(`verification error: ${JSON.stringify(phoneAuthSnapshot)}`))
            reject(new Error(`verification error: ${JSON.stringify(phoneAuthSnapshot)}`))
            break
          // android only events
          case firebase.auth.PhoneAuthState.AUTO_VERIFY_TIMEOUT:
            resolve(phoneAuthSnapshot)
            break
          case firebase.auth.PhoneAuthState.AUTO_VERIFIED:
            resolve(phoneAuthSnapshot)
            break
        }
      }, error => {
        reject(error)
      })
```

This is what I get as a response when I log phoneAuthSnapshot:
```
{ "code": null, "state": "error", "error": {} }
```

Does anybody know why this would work on a simulator but not on the actual device?

**Comment by mikehardy** [comment]
I have not seen this personally, last I checked phone auth was working in my work project and there are no other reports so this is in a kind of "not reproduced" state, and as a bug report it's missing all the details we would need to help troubleshoot.

**Reply by timurridjanovic** [reply]
I thought I had fixed the issue.. I realized that I was missing the APNs Authentication Key in firebase. I added it and it worked! But a few hours later it stopped working.. I have no clue why. Really strange..

**Reply by timurridjanovic** [reply]
but now even the captcha is failing on the simulator. I get this same response:
{ "code": null, "state": "error", "error": {} }

**Reply by timurridjanovic** [reply]
Now it started working again lol.. I guess it was a temporary issue with firebase?

**Reply by timurridjanovic** [reply]
stopped working again.. Could there be an issue if I'm using the same APNs auth key for 2 different apps? (It's the same app, just in a different env: development and prod).

**Reply by mikehardy** [reply]
Sorry - 

> I have not seen this personally, last I checked phone auth was working in my work project and there are no other reports so this is in a kind of "not reproduced" state, and as a bug report it's missing all the details we would need to help troubleshoot.
